### PR TITLE
Update csharp.md to indicate ConnectAsync is deprecated

### PR DESCRIPTION
**Pull request by aallbrig**
**What type of PR is this?**
/kind documentation

**What this PR does / Why we need it**:
As I was following the AgonesSDK C# guide I noticed that when I tried using `ConnectAsync` the function has actually been deprecated. It'd be good to get verbiage into the documentation so other Agones users know they don't have to use this method any longer.

Screenshot 2024-06-11 at 12 08 18 PM
Screenshot 2024-06-11 at 12 08 25 PM
Source @ [agones/sdks/csharp/sdk/AgonesSDK.cs#112](https://github.com/googleforgames/agones/blob/main/sdks/csharp/sdk/AgonesSDK.cs#L112)

**Which issue(s) this PR fixes**:

Closes #

**Special notes for your reviewer**:
Unsure if a strike through is desired or if the documentation block should just be completely removed. Let me know what you'd prefer. If I don't respond quickly enough please feel free to take on the documentation changes outside this PR.

### CODE REVIEW

**Comment by markmandel** [comment]
@igooch , @zmerlynn - given the new SDK compatibility standards, I _think_ we can delete this `ConnectAsync` function from the C# SDK, since it actually does nothing?

Or since it was implemented 9 months ago, we have to wait another 3 months?

**Comment by igooch** [comment]
We guarantee the SDK _Server_ compatibility, not the Client SDK implementation, so we're technically fine to remove it.

**Comment by markmandel** [comment]
> We guarantee the SDK _Server_ compatibility, not the Client SDK implementation, so we're technically fine to remove it.

Sounds like - this PR should delete the method entirely (label as `kind/breaking`) and update the SDK docs to [`feature` shortcode](https://agones.dev/site/docs/contribute/#within-a-page) hide the SDK method for the next version of Agones.

What do we think?

**Comment by zmerlynn** [comment]
> > We guarantee the SDK _Server_ compatibility, not the Client SDK implementation, so we're technically fine to remove it.
> 
> Sounds like - this PR should delete the method entirely (label as `kind/breaking`) and update the SDK docs to [`feature` shortcode](https://agones.dev/site/docs/contribute/#within-a-page) hide the SDK method for the next version of Agones.
> 
> What do we think?

Agreed.

**Comment by markmandel** [comment]
@aallbrig - you up for this change?

**Comment by aallbrig** [comment]
Yes!

**Comment by aallbrig** [comment]
@markmandel this change is ready to re-review!

File: site/content/en/docs/Guides/Client SDKs/csharp.md [hunk 1]
```diff
@@ -74,16 +74,17 @@ To use the AgonesSDK, you will need to import the namespace by adding `using Ago
 var agones = new AgonesSDK();
 ```
 
+{{% feature expiryVersion="1.42.0" %}}
 ### Connection
 
 To connect to the SDK server, either locally or when running on Agones, run the `ConnectAsync()` method.
 This will wait for up to 30 seconds if the SDK server has not yet started and the connection cannot be made,
```
Comment (igooch) [inline site/content/en/docs/Guides/Client SDKs/csharp.md:81]:
```suggestion
```

File: site/content/en/docs/Guides/Client SDKs/csharp.md [hunk 2]
```diff
@@ -74,16 +74,17 @@ To use the AgonesSDK, you will need to import the namespace by adding `using Ago
...
-and will return `false` if there was an issue connecting.
+and will return `false` if there was an issue connecting. This is an obsolete function that now always returns true.
```
Comment (igooch) [inline site/content/en/docs/Guides/Client SDKs/csharp.md:82]:
```suggestion
The `ConnectAsync()` method is an obsolete function that now always returns true.
```

**Review by igooch — COMMENTED** [review]
Overall looks good. There's a line towards the bottom of the csharp.md file `- All requests other than ConnectAsync will wait for up to 15 seconds before giving up, time to wait can also be set in the constructor.`. You can modify similar to how you did with {{% feature expiryVersion="1.42.0" %}}.

**Comment by aallbrig** [comment]
@igooch Added in suggested commits and added the expiry feature flag around ConnectAsync remark. Merged latest from main. Thanks for reviewing!

File: site/content/en/docs/Guides/Client SDKs/csharp.md [hunk 3]
```diff
@@ -358,7 +357,9 @@ bool isConnected = await agones.Alpha().IsPlayerConnectedAsync(playerId);
 ```
 
 ## Remarks
+{{% feature expiryVersion="1.42.0" %}}
 - All requests other than `ConnectAsync` will wait for up to 15 seconds before giving up, time to wait can also be set in the constructor.
+{{% /feature %}}
```
Comment (igooch) [inline site/content/en/docs/Guides/Client SDKs/csharp.md:362]:
```suggestion
{{% /feature %}}
{{% feature publishVersion="1.42.0" %}}
- All requests will wait for up to 15 seconds before giving up. Time to wait can also be set in the constructor.
{{% /feature %}}
```

**Review by igooch — COMMENTED** [review]
One last item then LGTM.

**Pull request merged**